superbench/common/utils/nvidia_helper.py

```diff
@@ -5,18 +5,185 @@
+    """
+    try:
+        handle = nvml.nvmlDeviceGetHandleByIndex(idx)
+        mem = nvml.nvmlDeviceGetMemoryInfo(handle)
+    except nvml.NVMLError as err:
+        logger.error('Get device memory failed: {}'.format(str(err)))
+        return None, None
+    return mem.used, mem.total
+
+
+def get_device_row_remapped_info(idx):
+    """Get the row remapped information of device.
+
+    The command 'nvidia-smi -i idx -q' contains the following output:
+        Remapped Rows
+            Correctable Error                 : 0
+            Uncorrectable Error               : 0
+            Pending                           : No
+            Remapping Failure Occurred        : No
+            Bank Remap Availability Histogram
+                Max                           : 640 bank(s)
+                High                          : 0 bank(s)
+                Partial                       : 0 bank(s)
+                Low                           : 0 bank(s)
+                None                          : 0 bank(s)
+        Temperature
+            GPU Current Temp                  : 36 C
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        remapped_metrics (dict): the row remapped information, None means failed to get the data.
+    """
+    output = run_command('nvidia-smi -i {} -q'.format(idx))
+    if output.returncode == 0:
+        begin = output.stdout.find('Remapped Rows')
+        end = output.stdout.find('Temperature', begin)
+        if begin != -1 and end != -1:
+            remapped_info = output.stdout[begin:end]
+            remapped_info = remapped_info.split('\n')
+            remapped_info = [item for item in remapped_info if ':' in item]
+            remapped_metrics = dict()
+            for item in remapped_info:
+                key_value = item.split(':')
+                key = 'gpu_remap_' + key_value[0].lower().strip().replace(' ', '_')
+                value = key_value[1].replace('bank(s)', '').strip()
+                try:
+                    value = int(value)
+                    remapped_metrics[key] = value
+                except BaseException:
+                    continue
+
+            return remapped_metrics
+
+    return None
+
+
+def get_device_ecc_error(idx):
+    """Get the ecc error information of device.
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        corrected_ecc (int)  : the count of single bit ecc error.
+        uncorrected_ecc (int): the count of double bit ecc error.
+    """
+    handle = nvml.nvmlDeviceGetHandleByIndex(idx)
+    corrected_ecc = 0
+    uncorrected_ecc = 0
+    for location_idx in range(nvml.NVML_MEMORY_LOCATION_COUNT):
+        try:
+            count = nvml.nvmlDeviceGetMemoryErrorCounter(
+                handle, nvml.NVML_MEMORY_ERROR_TYPE_CORRECTED, nvml.NVML_VOLATILE_ECC, location_idx
+            )
+            corrected_ecc += count
+        except nvml.NVMLError:
+            pass
```

Comment:
If it doesn't work, we cannot know it.

Reply:
Shall we implement it as get_device_memory?

Reply:
There have 8 types of memory location, for the nvml API design, the ecc counter is the total ecc error occurred in all the location:
```
class nvmlMemoryLocation_t(Enum):
    NVML_MEMORY_LOCATION_L1_CACHE = 0
    NVML_MEMORY_LOCATION_L2_CACHE = 1
    NVML_MEMORY_LOCATION_DRAM = 2
    NVML_MEMORY_LOCATION_DEVICE_MEMORY = 2
    NVML_MEMORY_LOCATION_REGISTER_FILE = 3
    NVML_MEMORY_LOCATION_TEXTURE_MEMORY = 4
    NVML_MEMORY_LOCATION_TEXTURE_SHM = 5
    NVML_MEMORY_LOCATION_CBU = 6
    NVML_MEMORY_LOCATION_SRAM = 7
    NVML_MEMORY_LOCATION_COUNT = 8
```
Different arch supports different locations, for example, A100 has `NVML_MEMORY_LOCATION_SRAM` supported but p100 doesn't. P100 has `NVML_MEMORY_LOCATION_TEXTURE_SHM` supported but A100 doesn't.

The error raised here is `NVMLError_NotSupported` for current device. The current implementation can ignore the device type and loop all the location to aggregate the ecc error.

Your suggestion is naming it as `get_device_memory`? Do you mean we can only get the ecc count from `NVML_MEMORY_LOCATION_DEVICE_MEMORY` ?

Reply:
Currently nvml.NVMLError uses `pass` and other type of exceptions will log error.